src/components/ui/DatePicker/DatePicker.svelte

```diff
@@ -0,0 +1,345 @@
+    }
+  }
+
+  function autoCompleteDate(event: Event) {
+    const { value } = getTarget(event);
+
+    const parsedDate = getDateFromString(`${value}`);
+    if (parsedDate !== null) {
+      setDateString(getDoyTime(parsedDate, parsedDate.getUTCMilliseconds() > 0));
+    }
+  }
+
+  function clearDate() {
+    setDateString('');
+  }
+
+  function closeDatePicker() {
+    isOpen = false;
+    isTouched = true;
+
+    dispatch('change', { value: dateString });
+  }
+
+  function decrementMonth() {
+    viewMonth = viewMonth - 1;
+  }
+
+  /**
+   * Converts a date string (YYYY-MM-DDTHH:mm:ss) or DOY string (YYYY-DDDDTHH:mm:ss) into a Date object
+   */
+  function getDateFromString(dateString: string): Date | null {
+    const parsedDate = parseDateTime(dateString);
+    if (parsedDate !== null) {
+      const { hour, min, ms, sec, year } = parsedDate;
+
+      if (isDoy(parsedDate)) {
+        return new Date(Date.UTC(year, 0, parsedDate.doy, hour, min, sec, ms));
+      }
+      return new Date(Date.UTC(year, parsedDate.month - 1, parsedDate.day, hour, min, sec, ms));
+    }
+
+    return null;
+  }
+
+  function incrementMonth() {
+    viewMonth = viewMonth + 1;
+  }
+
+  function isDoy(parsedDate: ParsedDateString | ParsedDoyString): parsedDate is ParsedDoyString {
+    return (parsedDate as ParsedDoyString).doy !== undefined;
+  }
+
+  /**
+   * Determines if a given string is in the correct date string format (YYYY-MM-DDTHH:mm:ss) or DOY string format (YYYY-DDDDTHH:mm:ss)
+   */
+  function isValidDateTime(dateString: string): boolean {
+    return parseDateTime(dateString) !== null;
+  }
+
+  function onChangeViewMonth({ detail }: CustomEvent) {
+    const { valueAsNumber } = getTarget(detail);
+    viewMonth = valueAsNumber;
+  }
+
+  function onChangeViewYear({ detail }: CustomEvent) {
+    const { valueAsNumber } = getTarget(detail);
+    viewYear = valueAsNumber;
+  }
+
+  function onDocumentKeydown(event: KeyboardEvent) {
+    if (event.key === 'Escape') {
+      closeDatePicker();
+    }
+  }
+
+  function onInputKeydown(event: KeyboardEvent) {
+    dispatch('keydown', event);
+  }
+
+  function onSelect({ detail }: CustomEvent) {
+    setDateString(getDoyTime(detail as Date, false));
+  }
+
+  function openDatePicker() {
+    isOpen = true;
+  }
+
+  function updateView(date: Date) {
+    viewMonth = date.getUTCMonth();
+    viewYear = date.getUTCFullYear();
+  }
+
+  function setDateString(value: string) {
+    dateString = value;
+
+    dispatch('change', { value });
+  }
+
+  function setToday() {
+    setDateString(getDoyTime(currentDate));
+  }
+</script>
+
+<div class="date-picker" bind:this={datePickerContainer}>
+  <input
+    class="st-input w-100"
+    class:error={!isValid || hasError}
+    {name}
+    autocomplete="off"
+    bind:value={dateString}
+    use:popperRef
+    on:change={autoCompleteDate}
+    on:click={openDatePicker}
+    on:focus={openDatePicker}
+    on:keydown={onInputKeydown}
```

Comment:
You should be able to DOM event forward this as well and remove `onInputKeydown`